models/productDao.js

```diff
@@ -16,11 +16,125 @@ const getProducts = async () => {
     JOIN location AS l ON p.location_id = l.id
     JOIN location_groups AS lg ON l.location_group_id = lg.id
     JOIN events AS e ON e.product_id = p.id
-    GROUP BY e.title;
+    GROUP BY e.title
   `);
   return getEventProduct;
 }
 
+const mainCategoryFiltering = async (mainCategoryName, sort) => {
+  const orderByobj = {
+    desc : '(p.price * (100 - p.discount_rate)) DESC',
+    asc : '(p.price * (100 - p.discount_rate)) ASC'
+  }
+  const descAndAsc = sort ? orderByobj[sort] : 'p.id';
+
+  const filtering = appDataSource.query(`
+    SELECT
+      lg.name AS locationGroupName,
+      p.thumbnail_image_url AS thumbnailImageUrl,
+      p.name,
+      p.price,
+      p.discount_rate 
+    FROM products AS p
+    JOIN sub_categories AS sc ON p.sub_category_id = sc.id
+    JOIN main_categories AS mc ON sc.main_category_id = mc.id
+    JOIN location AS l ON p.location_id = l.id
+    JOIN location_groups AS lg ON l.location_group_id = lg.id
+    WHERE mc.name = '${mainCategoryName}'
+    ORDER BY ${descAndAsc}  	
+    LIMIT 12 OFFSET 0
+  `);
+  return filtering;
+}
+
+const getProductMainCategories = async (mainCategoryName) => {
+  const productsMainCategories = await appDataSource.query(`
+    SELECT
+      mc.name AS mainCategoryName,
+      (SELECT 
+        JSON_ARRAYAGG(
+          JSON_OBJECT('name', sc.name)
+        ) 
+      FROM sub_categories AS sc
+      JOIN main_categories AS mc ON sc.main_category_id = mc.id
+      WHERE mc.name = '${mainCategoryName}'
+      ) AS subCategories,
+    JSON_ARRAYAGG(
+      JSON_OBJECT(
+        'locationGroupName', lg.name,
+        'thumbnailImageUrl', p.thumbnail_image_url,
+        'name', p.name,
+        'price', p.price,
+        'discountRate', p.discount_rate 
+      )
+    ) AS products
+    FROM products AS p
+    JOIN sub_categories AS sc ON p.sub_category_id = sc.id
+    JOIN main_categories AS mc ON sc.main_category_id = mc.id
+    JOIN location AS l ON p.location_id = l.id
+    JOIN location_groups AS lg ON l.location_group_id = lg.id
+    WHERE mc.name = '${mainCategoryName}'
+    LIMIT 12 OFFSET 0
+  `);
+  return productsMainCategories;
+}
+
+const subCategoryFiltering = async (mainCategoryName, subCategoryName, sort) => {
+  const orderByobj = {
+    desc : '(p.price * (100 - p.discount_rate)) DESC',
+    asc : '(p.price * (100 - p.discount_rate)) ASC'
+  }
+  const descAndAsc = sort ? orderByobj[sort] : 'p.id';
+
+  const filtering = appDataSource.query(`
```

Comment:
이 부분도 비동기 함수 처리를 위해 await 키워드를 넣어주세요!
```suggestion
  const filtering = await appDataSource.query(`
```